src/test/kotlin/org/jmailen/gradle/kotlinter/functional/ReportersTest.kt

```diff
@@ -0,0 +1,216 @@
+                    reporters = [
+                        'checkstyle',
+                        'html',
+                        'json',
+                        'plain',
+                        'sarif',
+                    ] 
+                }
+
+                """
+            appendText(buildScript)
+        }
+
+        build("lintKotlin").apply {
+            assertEquals(TaskOutcome.SUCCESS, task(":lintKotlin")?.outcome)
+            assertEquals(expectedEmptyPlain(), reportContent("/main-lint.txt"))
+            assertEquals(expectedEmptyCheckstyle(), reportContent("main-lint.xml"))
+            assertEquals(expectedEmptyHtml(), reportContent("/main-lint.html"))
+            assertEquals(expectedEmptyJson(), reportContent("/main-lint.json"))
+            assertTrue(reportContent("/main-lint.sarif.json").isNotBlank())
+        }
+        build("lintKotlin").apply {
+            assertEquals(TaskOutcome.UP_TO_DATE, task(":lintKotlin")?.outcome)
+        }
+
+        projectRoot.resolve("src/main/kotlin/CustomClass.kt") {
+            writeText(kotlinClass("WrongClassName"))
+        }
+
+        buildAndFail("lintKotlin").apply {
+            assertEquals(TaskOutcome.FAILED, task(":lintKotlinMain")?.outcome)
+            assertEquals(expectedFailedPlain(), reportContent("/main-lint.txt"))
+            assertEquals(expectedFailedCheckstyle(), reportContent("/main-lint.xml"))
+            assertEquals(expectedFailedHtml(), reportContent("/main-lint.html"))
+            assertEquals(expectedFailedJson(), reportContent("/main-lint.json"))
+            assertTrue(reportContent("/main-lint.sarif.json").isNotBlank())
+        }
+    }
+
+    @Test
+    fun `uses reporters from overridden ktlint version`() {
+        projectRoot.resolve("build.gradle") {
+            // language=groovy
+            val buildScript =
+                """
+
+                kotlinter {
+                    reporters = ['sarif']
+                     ktlintVersion = "0.47.0"
+                }
+
+                """
+            appendText(buildScript)
+        }
+
+        build("lintKotlin").apply {
+            val reportContent = projectRoot.resolve("build/reports/ktlint/main-lint.sarif.json").readText()
+            assertTrue(reportContent.contains(""""version": "0.47.0""""))
+            assertTrue(reportContent.contains(""""semanticVersion": "0.47.0""""))
+        }
+    }
+
+    private fun reportContent(reportName: String) = projectRoot.resolve("build/reports/ktlint/$reportName").readText()
+}
+
+private fun expectedEmptyPlain() = ""
+
+// language=xml
+private fun expectedEmptyCheckstyle() = """
+<?xml version="1.0" encoding="utf-8"?>
+<checkstyle version="8.0">
+</checkstyle>
+
+""".trimIndent()
+
+// language=html
+private fun expectedEmptyHtml() = """
+<html>
+<head>
+<link href="https://fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" />
+<meta http-equiv="Content-Type" Content="text/html; Charset=UTF-8">
+<style>
+body {
```

Comment:
Asserting on file content isn't necessarily the most reliable way to ensure proper behavior, and we probably should only compare smaller parts of the content. I wanted to avoid making _decisions_ what to test here specifically, and decided to start with full-content comparison and then migrate to either smaller chunks or start parsing the reports? Ideas are welcome

Reply:
Yeah, seems a little heavy. If we just check the content for a small known snippet, what's the real chance we'd have a false positive test pass?